tools/cli/services/create.js

```diff
@@ -1,25 +1,70 @@
-const fs = require('fs');
-const execute = require('../utils/execute');
-const { red } = require('../utils/colors');
-const { join } = require('path');
+const fs = require("fs");
+const execute = require("../utils/execute");
+const { red } = require("../utils/colors");
+const { join } = require("path");
+
+function runNpmInstallInDirectoryAndExitIfFailed(directoryName) {
+  try {
+    execute(
+      `npm install --prefix ${directoryName}`,
+      "npm install dependencies"
+    );
+  } catch {
+    console.log(red(`Failed to install dependencies using \`npm install\`.`));
+    process.exit(1);
+  }
+}
+
+function installPm2WithNpmIfMissingInDirectoryAndExitIfFailed(directoryName) {
+  if (isPm2NotExists()) {
+    installPm2WithNpmInDirectoryAndExitIfFailed(directoryName);
+  }
+}
+
+function isPm2NotExists() {
+  try {
+    execute(`which pm2`, `check if \`pm2\` is installed globally`);
+    return false;
+  } catch (err) {
+    return true;
+  }
+}
+
+function installPm2WithNpmInDirectoryAndExitIfFailed(directoryName) {
+  try {
+    execute(`npm install pm2 --prefix ${directoryName}`, "npm install pm2");
```

Comment:
1. Use cwd option install of prefix.
2. If it cannot be installed globally, we can find a workaround such as "alias pm2="npx pm2".
3. Add a e2e test for this case on the CLI. It's very important to make sure it's working.
4. The description should be something like "installing pm2".

Reply:
I'm not familiar with `pm2` or how it's used in the application. But by using an alias, wouldn't the alias be deleted once the user exits the terminal?

In addition, how can I make an e2e test? Suppose the user does have `pm2` installed globally, and in the test, I run the `greenpress create` command, how can I emulate a state in which the user doesn't have the package?

Reply:
The tests are running in a fresh VM for every PR and commit to the main branch.
Also, every test runs one by another, and not parallel, you can remove PM2 globally before the test case, and then run the test.